tests/ga/test_extension.py

```diff
@@ -1593,6 +1833,534 @@ def test_it_should_parse_boolean_values_encoded_as_strings(self):
+
+        data = '''{
+      "name": "ExampleHandlerLinux",
+      "version": 1.0,
+      "handlerConfiguration": {
+        "linux": {
+          "resources": {
+            "memory": {
+              "max_limit_percentage": 20,
+              "max_limit_MBs": "1000"
+            }
+          }
+        }
+      }
+    }
+    '''
+        handler_config = HandlerConfiguration(json.loads(data))
+        resource_config = handler_config.get_resource_configurations()
+
+        self.assertEqual(resource_config.get_cpu_limits_for_extension(), None)
+        self.assertEqual(resource_config.get_memory_limits_for_extension(), None)
+
+    def test_handler_configuration_incorrect_configuration(self, *args):
+        data = '''{
+          "name": "ExampleHandlerLinux",
+          "version": 1.0,
+          "handlerConfiguration": {
+            "linux": {
+              "resources": {
+              }
+            }
+          }
+        }
+        '''
+        with self.assertRaises(ExtensionConfigurationError) as context:
+            HandlerConfiguration(json.loads(data)).get_resource_configurations()
+
+        self.assertTrue("Incorrect resource configuration provided" in str(context.exception))
+
+        data = '''{
+                  "name": "ExampleHandlerLinux",
+                  "version": 1.0,
+                  "handlerConfiguration": {}
+                }
+                '''
+        with self.assertRaises(ExtensionConfigurationError) as context:
+            HandlerConfiguration(json.loads(data))
+
+        self.assertTrue("No linux configurations present in HandlerConfiguration" in str(context.exception))
+
+        data = '''{
+    "name": "ExampleHandlerLinux",
+    "version": 1.0
+    }
+    '''
+        with self.assertRaises(ExtensionConfigurationError) as context:
+            HandlerConfiguration(json.loads(data))
+
+        self.assertTrue("Malformed handler configuration file" in str(context.exception))
+
+    @patch("azurelinuxagent.ga.exthandlers.HandlerConfiguration.send_handler_configuration_event")
+    def test_load_handler_configuration(self, patch_send_handler_configuration_event, *args):
+        handler_name = "Not.A.Real.Extension"
+        handler_version = "1.2.3"
+
+        ext_handler = ExtHandler(handler_name)
+        ext_handler.properties.version = handler_version
+        ext_handler_i = ExtHandlerInstance(ext_handler, "dummy protocol")
+
+        with patch(
+                "azurelinuxagent.ga.exthandlers.ExtHandlerInstance.get_handler_configuration_file") as \
+                patch_get_handler_configuration_file:
+            patch_get_handler_configuration_file.return_value = load_data_path("ext/SampleHandlerConfiguration.json")
+            handler_config = ext_handler_i.load_handler_configuration()
+            self.assertNotEqual(handler_config, None)
+            self.assertEqual(patch_send_handler_configuration_event.call_count, 0)
```

Comment:
I thought `SampleHandlerConfiguration.json` was valid, why do we expect this isn't called?

Reply:
I am checking This is the telemetry event check. To make sure if telemetry is being sent for invalid cases.